galsim/chromatic.py

```diff
@@ -227,6 +460,152 @@ def drawImage(self, bandpass, image=None, integrator=None, **kwargs):
         image += integral
         return image
 
+    def _interp_drawImage(self, bandpass, image=None, integrator=None, **kwargs):
+        """Draw method adapted to work for ChromaticImage instances for which interpolation between
+        stored images is being used.  Users should not call this routine directly, and should
+        instead interact with the `drawImage` method.
+        """
+        # Check whether the object has had some chromatic transformation applied to it.  Since we
+        # are interpolating between stored images, this is not a situation that _interp_drawImage
+        # can handle.  We have to lose the interpolation.
+        if hasattr(self, '_A') and self._chromaticTransformation(bandpass):
+            import warnings
+            warnings.warn("Cannot render image with chromatic transformation applied to it"
+                          " using interpolation between stored images.  Reverting to "
+                          "non-interpolated version.")
+            self.noInterpolation()
+            return self._normal_drawImage(bandpass, image=image,
+                                          integrator=integrator, **kwargs)
+
+        # Note that integrator is not actually selectable.
+        if integrator is not None:
+            import warnings
+            warnings.warn("Cannot choose image integrator when using stored images! "
+                          "Ignoring this keyword argument.")
+
+        # setup output image (semi-arbitrarily using the bandpass effective wavelength).
+        # Note: we cannot just use self._imageAtWavelength, because that routine returns an image
+        # with whatever pixel scale was required to sample all the images properly.  We want to set
+        # up an output image that has the requested pixel scale, which might change the image size
+        # and so on.
+        prof0 = self._evaluateAtWavelength(bandpass.effective_wavelength)
+        image = prof0.drawImage(image=image, setup_only=True, **kwargs)
+        # Remove from kwargs anything that is only used for setting up image:
+        kwargs.pop('dtype', None)
+        kwargs.pop('scale', None)
+        kwargs.pop('wcs', None)
+        kwargs.pop('nx', None)
+        kwargs.pop('ny', None)
+
+        # determine combined self.wave_list and bandpass.wave_list
+        wave_list = self._getCombinedWaveList(bandpass)
+
+        # merge self.wave_list into bandpass.wave_list since we are using a sampling integrator
+        bandpass = galsim.Bandpass(galsim.LookupTable(wave_list, bandpass(wave_list),
+                                                      interpolant='linear'))
+
+        # The integration is carried out using the following two basic approaches:
+        # (1) We use linear interpolation between the stored images to get an image at a given
+        #     wavelength.
+        # (2) We use the midpoint rule for integration.
```

Comment:
I think it's probably worthwhile.  Trapezoid really is quite a bit more accurate than midpoint for a given number of samples.  And since the point of this is to be able to cut down the number of samples for the integration, it seems worth it to use trapezoid, where that number can generally be made even smaller.